### PR TITLE
[rpm] Improve Mer/SailfishOS  RPM spec file for EmbedLite 38.

### PR DESCRIPTION
**Pull request by tworaz**
Apart from some cleanup this actually makes it possible to build the
engine for Mer/SailfishOS. The current spec does not work anymore.

### CODE REVIEW

**Comment by rojkov** [comment]
LGTM

File: rpm/xulrunner-qt5.spec [hunk 1]
```diff
@@ -1,8 +1,42 @@
-%define greversion 31.0a1
+%define greversion  38.0.5
+%define patch_level 0
+
+%define embedlite_config merqtxulrunner
+
+%define system_sqlite     0
```
Comment (rainemak) [inline rpm/xulrunner-qt5.spec:6]: Any reason why not using system sqlite?

Comment (tworaz) [inline rpm/xulrunner-qt5.spec:6]: Mer currently ships sqlite 3.8.5, gecko 38 minimum requirement is 3.8.9.

Comment (rainemak) [inline rpm/xulrunner-qt5.spec:6]: My point exactly. That should be fixed, see PR https://github.com/mer-packages/sqlite/pull/5

File: rpm/xulrunner-qt5.spec [hunk 2]
```diff
@@ -12,13 +46,14 @@ BuildRequires:  pkgconfig(Qt5Quick)
 BuildRequires:  pkgconfig(Qt5Network)
```
Comment (rainemak) [inline rpm/xulrunner-qt5.spec:46]: Could you drop `BuildRequires: pkgconfig(Qt5Quick)` at the same for when improving spec.

**Comment by rainemak** [comment]
LGTM. Let's keep the system sqlite disabled for now and further cleanups can be done later.

**Comment by rojkov** [comment]
The spec looks saner now.

**Pull request merged**